packages/testing/src/fixture.ts

```diff
@@ -0,0 +1,188 @@
+import { existsSync } from 'fs'
+import { createRequire } from 'module'
+import { normalize } from 'path'
+import { fileURLToPath } from 'url'
+
+import test from 'ava'
+import cpy from 'cpy'
+import { execa, execaCommand } from 'execa'
+import stringify from 'fast-safe-stringify'
+import { getBinPathSync } from 'get-bin-path'
+import isPlainObj from 'is-plain-obj'
+
+import { createRepoDir, removeDir } from './dir.js'
+import { ServerHandler, startServer, Request } from './server.js'
+
+const require = createRequire(import.meta.url)
+
+export class Fixture {
+  flags: Record<string, unknown> = {
+    stable: true,
+    buffer: true,
+    branch: 'branch',
+    featureFlags: {},
+  }
+
+  env = {
+    // Ensure local environment variables aren't used during development
+    // TODO: check this one
+    // BUILD_TELEMETRY_DISABLED: '',
+    NETLIFY_AUTH_TOKEN: '',
+  }
+
+  copyRootDir: string
+  /** The absolute path of the fixtures repository root */
+  repositoryRoot = ''
+
+  /** The binary of @netlify/build */
+  private buildBinaryPath = getBinPathSync({ cwd: require.resolve('@netlify/build') })
+  private configBinaryPath = getBinPathSync({ cwd: require.resolve('@netlify/config') })
+
+  constructor(
+    /**
+     * a relative path from the test file to the fixture directory
+     * @example: ./fixtures/my_fixture
+     */
```

Comment:
This will be easier to read with the comment on top of constructor definition.

Reply:
But then intellisence will not pick it up. I add the type to the constructor prop directly which will result in providing the doc during usage:

<img width="766" alt="CleanShot 2022-09-26 at 13 12 17@2x" src="https://user-images.githubusercontent.com/11156362/192262447-15dd7951-1206-4a49-a7f6-fa115a927ec7.png">

especially relevant when you have multiple params and you want to just document each param